packages/core/src/core.ts

```diff
@@ -651,18 +652,24 @@ class ThresholdKey implements ITKey {
   }: {
     determinedShare?: BN;
     initializeModules?: boolean;
-    importedKey?: BN;
+    importedKey?: Buffer;
     delete1OutOf1?: boolean;
   } = {}): Promise<InitializeNewKeyResult> {
-    let seed: Uint8Array = new Uint8Array();
+    let seed: Buffer;
     if (this.keyType === KeyType.secp256k1) {
-      const tmpPriv = importedKey || generatePrivate(this.keyType);
+      const tmpPriv = importedKey ? new BN(importedKey) : generatePrivate(this.keyType);
       this._setKey(new BN(tmpPriv));
     } else if (this.keyType === KeyType.ed25519) {
-      seed = importedKey ? new Uint8Array(importedKey.toBuffer(undefined, 32)) : nacl.randomBytes(32);
-      const keyPair = nacl.sign.keyPair.fromSeed(seed);
-      const tempPriv = new BN(keyPair.secretKey.slice(0, 32)).umod(this.ecCurve.curve.n);
-      this._setKey(tempPriv);
+      seed = importedKey && importedKey;
+      if (!seed) {
+        const newEd25519Seed = await getRandomBytes(32);
+        seed = Buffer.from(newEd25519Seed);
+      }
+      const keyPair = getEd25519ExtendedPublicKey(seed);
+      // note this scalar is litte endian encoded
+      // no need of converting to BE, since we only use this for encryption and decryption.
+      const ed25519Scalar = keyPair.scalar;
+      this._setKey(ed25519Scalar);
```

Comment:
I don't understand the following:

My understanding is we use secp256k1-based encryption.
So the encryption key should be an secp256k1 scalar.
But now we set the key to be a ed25519 scalar.

Am I missing something?

Reply:
So, what we do it shares are created using ed25519 scalar only and while enc and dec on metadata layer (in storage layer class) we convert this scalar to secp. Although i havent seen any use case of deriving shares from ed25519 scalar, it will work even if we use secp shares, only benefit of using ed25519 scalar is we can check the integrity of public key being derived from encrypted seed, that should exactly match the pub key being derived from this scalar. Also not sure if any future usecase will require ed25519 shares so kept it this way. @ieow can provide more details